<launch>


    <node pkg = "rtabmap_ros"
          type = "rtab_broadcaster.py"
          name = "rtab_broadcaster_node"
          args = ""
          output = "screen">
    </node>


    <node pkg = "rtabmap_ros"
          type = "topic_repub.py"
          name = "rtab_repub_node"
          args = "_topics:='Imu        /imu/data                                 /rtabmap/imu                  rtab_imu
                            Image      /d435i/aligned_depth_to_color/image_raw   /rtabmap/rgb/image_raw        rtab_rgb
                            CameraInfo /d435i/aligned_depth_to_color/camera_info /rtabmap/rgb/camera_info      rtab_rgb
                            Image      /d435i/depth/image_rect_raw               /rtabmap/depth/image_rect_raw rtab_depth
                            CameraInfo /d435i/depth/camera_info                  /rtabmap/depth/camera_info    rtab_depth'"
          output = "screen"/>


    <include file = "$(find rtabmap_ros)/launch/rtabmap.launch" if="true">

        <arg name = "rgb_topic"               value = "/rtabmap/rgb/image_raw"/>
        <arg name = "depth_topic"             value = "/rtabmap/depth/image_rect_raw"/>
        <arg name = "camera_info_topic"       value = "/rtabmap/rgb/camera_info"/>
        <arg name = "depth_camera_info_topic" value = "/rtabmap/depth/camera_info"/>

        <arg name = "args"                    value = "--delete_db_on_start
                                                       --Optimizer/GravitySigma 0.3
                                                       --Odom/Holonomic false
                                                       --Odom/ResetCountdown 1
                                                       --Rtabmap/StartNewMapOnLoopClosure true
                                                       --Odom/Strategy 1"/>

        <arg name = "queue_size"              value = "10"/>
        <arg name = "approx_sync"             value = "true"/>
        <arg name = "wait_imu_to_init"        value = "false"/>
        <arg name = "visual_odometry"         value = "true"/>

        <arg name = "odom_topic"              value = "/rtabmap/odom"/>
        <arg name = "imu_topic"               value = "/rtabmap/imu"/>
        <arg name = "map_frame_id"            value = "rtab_init"/>
        <arg name = "frame_id"                value = "rtab_odom"/>
        <arg name = "use_sim_time"            value = "true"/>
        <arg name = "rtabmapviz"              value = "true"/>
        <arg name = "rviz"                    value = "false"/>

    </include>


</launch>
